modules/client-handler/src/integrationTest/java/org/apache/ignite/client/handler/ItSslClientHandlerTest.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.client.handler;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import io.netty.handler.ssl.util.SelfSignedCertificate;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.Socket;
+import java.net.SocketException;
+import java.nio.file.Path;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.io.TempDir;
+import org.msgpack.core.MessagePack;
+
+/** SSL client integration test. */
+public class ItSslClientHandlerTest {
+
+    /** Magic bytes. */
+    private static final byte[] MAGIC = {0x49, 0x47, 0x4E, 0x49};
+
+    private ClientHandlerModule serverModule;
+
+    private TestServer testServer;
+
+    private int serverPort;
+
+    private String keyStorePkcs12Path;
+
+    @BeforeEach
+    void setUp(@TempDir Path tmpDir) throws Exception {
```

Comment:
We don't use `@TempDir` anywhere and usually use `@WorkDirectory` instead

Reply:
But I thought that `WorkDirectory` is used as a working directory for Ignite node.  Why can't we use the standard unit 5 extension to store some temporary configuration files?

Reply:
Just for consistency. The only difference between `TempDir` and `WorkDirectory` is that you can keep the generated folder for debugging purposes

Reply:
No problem will use WorkDirectory